### DISCUSSION TITLE
Tesla Developer - error "Invalid Domain. Domain must be registered with a certificate authority."

### DISCUSSION THREAD

**Discussion by sgryphon** [llamafilm/tesla-http-proxy-addon#128] [General]
I had some trouble registering the Telstra Developer app -- I emailed their support, and haven't heard back yet, but did keep working on it and found a work around.

They seem to have some limitations on which domains they accept (even if the domains are valid, have valid certs, etc.

BTW. The best cert checker I found was Qualys/SSL Labs, i.e. https://www.ssllabs.com/ssltest/analyze.html?d=tesla.gamertheory.net

All the following had Let's Encrypt certs, but only 'www' worked.

* `tesla.gamertheory.net` -- didn't work, note that the host is in the SAN (subject alternative name), not the CN (common name)
* `tesla-auth.gamertheory.net` -- didn't work, for this on the host was the common name
* `www.gamertheory.net` -- did work (host was common name)

**Comment by sgryphon** [comment]
I was able to complete my developer application, and it immediately said approved (and sent me an email saying approved).

I was able to log in, paste the callback URL, and get the refresh token; and also the test public key endpoint works.

However, when I try to enroll public key in your vehicle I get the error:

"Unable to Share Vehicle. This third party isn't registered with Tesla. We can't grant them access at this time."

I guess this could be a timing thing, i.e. while my app is registered it has been loaded into all their systems yet, so I will try again later.

**Comment by Aephir** [comment]
Did you ever hear back? I have the same problem, and nothing seems to work, I've quite a few (incl. `www`), and all give the same error:
```
Invalid Domain. Domain must be registered with a certificate authority and cannot contain "Tesla".
```

**Reply by karel1980** [reply]
Same here. Valid Letsencrypt certificate. I'm running this from home and my ISP doesn't allow incoming traffic to well known ports, so I'm running https on port 8443. I thought maybe could be the problem, idk.

**Comment by jsammarco** [comment]
ONE More EDIT: NVM again, had the return url in the wrong field. It also does not like 2 fields for the same value so once I removed the extra and put the return url in it works!!!

EDIT AGAIN: NVM. it's doing it again for no reason. Can't save

EDIT: Now all of a sudden its working. Teslas CA check service must have been down.

WTH. Same here. It let me create the app with the domain but I can't update the app because I get the same error.
Tesla get ur sh*t together.
